remove enabling of file sharing and creation of smbshare from MSI

anjannath opened the pull request:
fixes #4619

anjannath committed:
remove enabling of file sharing and creation of smbshare from MSI

this will be moved to the 'crc setup' command as we should only enable
file sharing and create the smb share for home dir when users have set
the config option 'enable-shared-dirs' to 'true'

this should remove at least of the warnings reported by virustotal for
the MSI:
https://www.virustotal.com/gui/file/31b402dcc1da24265074a21a26018d6cde8eef0b63c77a18f89eb079b6556790

anjannath committed:
preflight: add new label SharedDir for checks

this label will be used to filter out preflight checks
that should only run when 'enable-shared-dir' settings
is set to 'true'

anjannath committed:
preflight: add check, fix and cleanup funcs for enabling SMB share on windows

earlier these were part of the MSI since we want to enable this only
when user has set 'enable-shared-dirs' setting, this is moved to the
preflight package where we can check this config value before hand

this should remove at least of the warnings reported by virustotal
for the MSI

https://www.virustotal.com/gui/file/31b402dcc1da24265074a21a26018d6cde8eef0b63c77a18f89eb079b6556790

praveenkumar commented:
@anjannath didn't we have this as part of preflight before and then moved it to MSI? if yes then we might to revert those commits instead of this?

anjannath commented:
> @anjannath didn't we have this as part of preflight before and then moved it to MSI? if yes then we might to revert those commits instead of this?

no, i also thought the same and i did check the logs before creating the PR, seems this was never added as preflight checks, i think because when we added this SMB sharing support we were already moving every privileged operation on windows already to MSI so when we added support for SMB directory sharing this was only added to the MSI..

```
$ git log --grep="smb" --grep="cifs" --oneline
fff474a33 msi: remove the smb share created during installation
2aa7059b4 msi: ignore errors from custom action creating the smb shared dir
4a52aef4e cifs: Add 'enable-shared-dirs' config for windows
b4cfc0ec5 add CIFS/SMB based file sharing for windows
2785de06d move initialization of SharedDirs to driver's CreatHost function
````

cfergeau commented on pkg/crc/preflight/preflight_darwin_test.go:
```diff
@@ -17,9 +17,9 @@ func TestCountConfigurationOptions(t *testing.T) {
-	assert.Len(t, getPreflightChecks(true, network.UserNetworkingMode, constants.GetDefaultBundlePath(preset.OpenShift), preset.OpenShift, false), 20)
-	assert.Len(t, getPreflightChecks(true, network.UserNetworkingMode, constants.GetDefaultBundlePath(preset.OpenShift), preset.OpenShift, false), 20)
+	assert.Len(t, getPreflightChecks(true, network.UserNetworkingMode, constants.GetDefaultBundlePath(preset.OpenShift), preset.OpenShift, false, false), 20)
+	assert.Len(t, getPreflightChecks(true, network.UserNetworkingMode, constants.GetDefaultBundlePath(preset.OpenShift), preset.OpenShift, false, false), 20)
```
There are several changes to `getPreflightChecks` calls in this commit to add one argument, they seem to belong to the previous commit.

anjannath replied:
updated

cfergeau commented on packaging/windows/product.wxs.template:
```diff
@@ -92,24 +92,6 @@
         <CustomAction Id="AddUserToHypervAdminGroup" BinaryKey="WixCA" DllEntry="WixQuietExec64" Execute="deferred" Impersonate="no" Return="ignore" />
-        <SetProperty Action="CACreateSMBShare"
-            Id="CreateSMBShare"
-            Value="&quot;[POWERSHELLEXE]&quot; -NonInteractive -ExecutionPolicy Bypass -NoProfile -Command &quot;New-SmbShare -Name '[SHAREDDIRNAME]' -Path '[USERFOLDER]' -FullAccess '[LogonUser]'&quot;"
```
I guess you should also be removing 
```
        <Property Id="USERFOLDER">
            <DirectorySearch Id="userProfileSearch" Depth="0" Path="[%USERPROFILE]" />
        </Property>

        <Property Id="SHAREDDIRNAME" Secure="yes">crc-dir0</Property>
```
and possibly other properties? (powershell, logonuser, …)

anjannath replied:
we need to keep powershell and logonuser these are used by the installHyperv custom action

cfergeau reviewed (commented):
> this should remove at least of the warnings reported by virustotal for
the MSI:

One concern I have is if virustotal starts reporting this against the crc binary instead of reporting that against the msi

redbeam commented:
@anjannath I suggest closing this as I opened a separate PR that removes SMB file sharing support completely (#4879).

anjannath commented:
closed in favor of #4879
